File: 프로그래머스/lv1/118666. 성격 유형 검사하기/성격 유형 검사하기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(vector<string> survey, vector<int> choices) {
    string answer = "";
    int rt = 0, cf = 0, jm = 0, an = 0;
    for (int i = 0; i < survey.size(); i++) {
        if (survey[i] == "AN") an += choices[i] - 4;
        else if (survey[i] == "NA") an -= (choices[i] - 4);
        else if (survey[i] == "RT") rt += (choices[i] - 4);
        else if (survey[i] == "TR") rt -= (choices[i] - 4);
        else if (survey[i] == "CF") cf += (choices[i] - 4);
        else if (survey[i] == "FC") cf -= (choices[i] - 4);
        else if (survey[i] == "JM") jm += (choices[i] - 4);
        else if (survey[i] == "MJ") jm -= (choices[i] - 4);
    }
    (rt > 0) ? answer += "T" : answer += "R";
    (cf > 0) ? answer += "F" : answer += "C";
    (jm > 0) ? answer += "M" : answer += "J";
    (an > 0) ? answer += "N" : answer += "A";
    return answer;
}

File: 프로그래머스/lv1/118666. 성격 유형 검사하기/성격 유형 검사하기.py
def solution(survey, choices):
    answer = ''
    rt, cf, jm, an = 0, 0, 0, 0
    for i in range(len(survey)) :
        if (survey[i] == "AN") :
            an += choices[i] - 4
        elif (survey[i] == "NA") :
            an -= (choices[i] - 4)
        elif (survey[i] == "RT") :
            rt += (choices[i] - 4)
        elif (survey[i] == "TR") :
            rt -= (choices[i] - 4)
        elif (survey[i] == "CF") :
            cf += (choices[i] - 4)
        elif (survey[i] == "FC") :
            cf -= (choices[i] - 4)
        elif (survey[i] == "JM") :
            jm += (choices[i] - 4)
        else :
            jm -= (choices[i] - 4)
    answer += "T" if rt > 0 else "R"
    answer += "F" if cf > 0 else "C"
    answer += "M" if jm > 0 else "J"
    answer += "N" if an > 0 else "A"
    return answer